### ISSUE TITLE
Any plans to support .NET Standard 2.1 (and ASP.Net Core 3.1)?

### ISSUE THREAD

**Issue by Stanley-G** [JakeGinnivan/WebApi.Hal#155]
I am porting a .NET Core web application from ASP.NET Core 2.2 to 3.1, and discovered that the `WebApi.Hal` library does not appear to support .NET Core 3.1, at least not yet.

This is in part because Microsoft has dropped their dependency on Newtonsoft.Json in favor of their own `System.Text.Json` library.

Are there plans to support .NET Standard 2.1 in the near future?

**Comment by panmanphil** [comment]
This was news to me, thanks for pointing it out. I've battled the pinned dependency on Newtonsoft.Json with .net functions, so this is welcome news. "We", as in the loosely organized maintainers in our part time support of the project have had no plans here. Pull requests are very welcome with the caveat that we will need to support netstandard 2 also (I think) so may need more build targets.

**Comment by johannesrittmeier** [comment]
Hi, past the holydays i was a bit bored, so i allready migrated WebApi.Hal to Core31.
The current version is not working with 31 because they changed some Formatter APIs.
This was the easy part (as far as i can trust the unittests)

Then in started to migrate to system.text.json and this is a very heavy task.
system.text.json isnt supporting the serialization attributes, it ignores inherited attributes and at last in only serializes properties of the type thats supplied.
At first i would avoid the migration to system.text.json.

is there a chance, that i could provide you my 3.1 migration?

Regards Johannes

**Comment by gulzarahamed** [comment]
we have found when the startup.cs class seperated and moved from hosting project this issue get resolved.

**Comment by dawiduniec** [comment]
Hello, what's the current status? When do you plan to support core 3.1?

**Comment by panmanphil** [comment]
We'll review this, hopefully tonight. Thanks so much for the effort!

**Comment by dawiduniec** [comment]
I have tested code from branch 3.1 and it doest work as expected. It does serialize _link no matter if it's application/json or application/hal+json

**Comment by panmanphil** [comment]
Reading through the details of the error running under dotnet core 3.1, ugh! I will ran a branch with the sample project updated to 3.1 and see what ideas I can come up with. Open to any ideas that still include supporting .net framework users.

**Comment by panmanphil** [comment]
TLDR; I think its a good idea to merge the PR to support .net core 3.1.

We'd accept the PR into a new version 4 that will temporarily drop .net framework users and only support .net core 3.1. .net framework users 4.8 and below and .net core users < 3.1 can continue to use the 3.1 version, and there is no new functionality anyway. The IsMultiLink PR I would accept into this 4.0 branch as it is also a potentially breaking change. Then when .net 5.0 comes out, we converge to a new version 5 that supports the current (then) netstandard2.1 and works in .net core and .net framework. 

Other ideas in range of more to less desirable:
- move the formatters to a separate projects, one for netstandard2.0 and one for .net core 3.1 and update docs to tell people how to wire these up. Would also need to find an alternative to the `[JsonIgnore]`. Seems like too much work, and worse usability for developers
- rewrite the formatters from scratch instead of basing them on the existing and now removed formatters form asp.net mvc. I don't think anybody wants to take that on, and while better for developers, could easily introduce breaking changes in a project that has no new functionality and works against a spec that is also not changing.

**Comment by lankymart** [comment]
Any update on this?

**Comment by panmanphil** [comment]
Since I've heard no other comments positive or negative, this is the way I'll go. Now when is the next question, in crunch time with my paid gig! Soon. But if the people that submitted their PR's want to update them, I'd be very ok with that too.

**Comment by panmanphil** [comment]
This has been updated and merged. A new release will be made and pushed to nuget once documentation has been updated

**Comment by lankymart** [comment]
Just tried the latest and it still has some dependencies dependent on .Net Framework which is causing it to error.

**Comment by panmanphil** [comment]
Just upload version 4.0.1 to nuget. This is the .net core 3.1 version. Thanks for the reminder
